dual_net.py

```diff
@@ -173,10 +199,137 @@ def train(self, boards, pi, z):
         should match input_shape and action_size as set during initialization.
         returns the batch loss
         """
+        self.boards = [state_to_board(board) for board in boards]
         self.sess.run([self.update_op], feed_dict={self.board_placeholder: boards,
                                                    self.pi: pi,
                                                    self.z: z})
         loss = self.sess.run([self.loss], feed_dict={self.board_placeholder: boards,
                                                      self.pi: pi,
                                                      self.z: z})
         return loss
+
+# Map from piece to layer in net input.  0-5 are white.
+INDEX_TO_PIECE_MAP = {0: chess.KING, 6: chess.KING,
+                      1: chess.QUEEN, 7: chess.QUEEN,
+                      2: chess.ROOK, 8: chess.ROOK,
+                      3: chess.BISHOP, 9: chess.BISHOP,
+                      4: chess.KNIGHT, 10: chess.KNIGHT,
+                      5: chess.PAWN, 11: chess.PAWN}
+
+KQK_INDEX_TO_PIECE_MAP = {0: chess.KING,
+                          1: chess.QUEEN,
+                          2: chess.KING}
+
+CHAR_TO_INDEX_MAP = {'K': 0, 'k': 6,
+                     'Q': 1, 'q': 7,
+                     'R': 2, 'r': 8,
+                     'B': 3, 'b': 9,
+                     'N': 4, 'n': 10,
+                     'P': 5, 'p': 11}
+
+
+def state_to_board(state):
```

Comment:
This is for the full chess environment, right? Not the KQK one? Maybe it deserves its own `ChessEnv` class?

Reply:
Yeah, I'm leaving it there until we actually make that.

Reply:
I pushed a rename to make the thing we have now the KQKChessEnv. What's stopping us from moving this out into its own ChessEnv class / module?